s3/pkg/service/object.go

```diff
@@ -355,45 +357,77 @@ func (s *s3Service) GetObject(ctx context.Context, req *pb.GetObjectInput, strea
 		return err
 	}
 
-	var finalReader io.Reader
-	finalReader = reader
+	allBytes, _ := ioutil.ReadAll(reader)
+
+	log.Infof("allBytes.len - %v", len(allBytes))
+
+	allDecBytes := make([]byte, 0)
+
 	if bucket.ServerSideEncryption.SseType == "SSE" {
-		finalReader, _ = utils.WrapAlignedEncryptionReader(reader, 0,
-			bucket.ServerSideEncryption.EncryptionKey,
-			bucket.ServerSideEncryption.InitilizationVector)
-	}
+		// this should match the upload chunk size during multipart upload, which is 5MB (5*1024*1024). During multipart
+		// upload, each part gets encrypted and uploaded, so here, we need to decrypt each part and re-assemble the file
+		chunkSize := 5242880
 
-	eof := false
-	left := object.Size
-	buf := make([]byte, ChunkSize)
-	for !eof && left > 0 {
-		n, err := finalReader.Read(buf)
-		if err != nil && err != io.EOF {
-			log.Errorln("failed to read, err:", err)
-			break
-		}
-		// From https://golang.org/pkg/io/, a Reader returning a non-zero number of bytes at the end of the input stream
-		// may return either err == EOF or err == nil. The next Read should return 0, EOF.
-		// If err is equal to io.EOF, a non-zero number of bytes may be returned.
-		if err == io.EOF {
-			log.Debugln("finished read")
-			eof = true
+		allBytesLen := len(allBytes)
+		modVal := allBytesLen / chunkSize
+
+		for i := 0; i < modVal; i++ {
+			decThouBytes := make([]byte, 0)
+			for j := (i) * chunkSize; j < (i+1)*chunkSize; j++ {
+				decThouBytes = append(decThouBytes, allBytes[j])
+			}
+
+			// encrypt in chunks of chunkSize, write file to disk
+			var out bytes.Buffer
+			block, err := aes.NewCipher(bucket.ServerSideEncryption.EncryptionKey)
```

Comment:
Hi @rhsakarpos , here use sse configuration of bucket, then what if a user enable sse of a bucket first, and upload an object, then disable the sse of the bucket, and download the object?

Reply:
that will fail. currently, we do not have disable SSE on bucket for this reason.